lib/rucio/core/rule.py

```diff
@@ -3365,11 +3380,11 @@ def __create_rule_approval_email(rule: "models.ReplicationRule", *, session: "Se
 
 
 @transactional_session
-def _create_recipients_list(rse_expression: str, filter_=None, *, session: "Session"):
+def _create_recipients_list(rse_expression: str, filter_: Optional[str] = None, *, session: "Session") -> list[tuple[str, InternalAccount]]:
```

Comment:
Annoyingly, in the current implementation, `InternalAccount` should be `Union[str, InternalAccount]`.